lambda-notes/src/Views/EditNoteContainer.js

```diff
@@ -0,0 +1,280 @@
+    border:1px solid #CECECE;
+    display:flex;
+`
+const NoteSection = styled.div`
+    width:75%;
+    background-color: #F2F1F2;
+    padding-left:25px;
+    padding-right:25px;
+    padding-top:25px;
+`
+const NavButton = styled.div`
+    background-color:#2FB8BC;
+    width:150px;
+    height:35px;
+    color:white;
+    display:flex;
+    justify-content:center;
+    align-items:center;
+`
+const FormSection = styled.div`
+    padding-left:25px;
+    padding-right:25px;
+
+`
+
+const Form = styled.form`
+    display:flex;
+    flex-direction:column;
+`
+
+const TitleInput = styled.input`
+    width:355px;
+    height:43px;
+`
+const TextArea = styled.textarea`
+    margin-top:25px;
+    width:610px;
+    height:350px;
+`
+const SaveButton = styled.button`
+    width:190px;
+    height:40px;
+    margin-top:15px;
+    background-color:#2FB8BC;
+    color:white;
+    font-size:1.7rem;
+    margin-bottom:100px;
+`
+const DeleteButton = styled.button`
+    background-color:#C3001F;
+    width:190px;
+    height:40px;
+    color:white;
+    font-size:1.7rem;
+    margin-right:10px;
+`
+const CancelButton = styled.button`
+    width:190px;
+    height:40px;
+    background-color:#2FB8BC;
+    color:white;
+    font-size:1.7rem; 
+    margin-left:10px;   
+`
+const WarningText = styled.p`
+    font-size:1.6rem;
+`
+const ModalActions = styled.div`
+    display:flex;
+    justify-content:space-around;
+`
+const FormTitle = styled.p`
+    margin-top:50px;
+    font-weight:bold;
+    font-size:2.3rem;
+    margin-bottom:30px;
+`   
+const EditLink = styled.p`
+    text-decoration:underline;
+    font-size:1.6rem;
+    margin-right:10px;
+    cursor:pointer;
+`
+const DeleteLink = styled.p`
+    text-decoration:underline;
+    font-size:1.6rem;
+    margin-left:10px;
+    cursor:pointer;
+`
+const NoteTitle = styled.p`
+    margin-top:50px;
+    font-weight:bold;
+    font-size:2.3rem;
+    margin-bottom:30px;    
+`
+const NoteBody = styled.p`
+    font-size:1.6rem;
+`
+class EditNoteContainer extends React.Component{
+    constructor(){
+        super()
+        this.state = {
+            isEditting:false,
+            id:'',
+            title:'',
+            body:'',
+            showModal:false
+        }
+    }
+    handleEditClick =event =>{
+        this.setState({
+            body:this.props.data.textBody,
+            title:this.props.data.title,
+            id:this.props.data._id,
+            isEditting:true
+        })
+    }
+    handleTextChange = event =>{
+        this.setState({[event.target.id]: event.target.value})
+    }
+    handleSubmit = event =>{
+        event.preventDefault()
+        this.props.editNote(this.state.id,{
+            textBody:this.state.body,
+            title:this.state.title
+        },
+
+        ()=>{
+            this.setState({
+                body:'',
+                title:'',
+                id:'',
+                isEditting:false
+            },()=>{this.props.history.push('/')})
```

Comment:
unclear why you're bothering to setState here if you immediately will go to a different view (and therefore this component will unmount...)